releasenotes/notes/sorted-circuit-parameters-7607dd7ef4e6cc0e.yaml

```diff
@@ -0,0 +1,18 @@
+---
+features:
+  - |
+    Return unbound parameters in a circuit in an order sorted by insertion and
+    allow to bind/assign parameters by a list of values or new parameters instead of
+    only by a dictionary. For example
+
+    .. parsed-literal::
+
+      from qiskit.circuit import QuantumCircuit, Parameter
+
+      circuit = QuantumCircuit(1)
+      circuit.rx(Parameter('x'), 0)
+      circuit.ry(Parameter('y'), 0)
+
+      print(circuit.parameters)  # ParameterView([x, y])
+
+      bound = circuit.bind_parameters([1, 2])
```

Comment:
Can you add a deprecation notice for the set-like operations on `QuantumCircuit.parameters`?

Reply:
other deprecated elements are missing, like `value_dict`.

Reply:
Good point, I added a notice for `value_dict` and `param_dict`